middleware/checkfor.js

```diff
@@ -1,10 +1,23 @@
-module.exports = (props) => (req, res, next) =>
-  props.map((prop) =>
-    req.body[prop]
-      ? next()
-      : res
-          .status(400)
-          .json({
-            errorMessage: `The ${prop} is required to create a video. Please include a ${prop} and try again.`,
-          })
-  );
+module.exports = (props) => (req, res, next) => {
+  let presentParams = [];
+  let missingParams = [];
+  let requestValues = Object.keys(req.body);
+
+  props.map((bodyParam) => {
+    if (requestValues.includes(bodyParam)) {
+      presentParams = [...presentParams, bodyParam];
+    } else {
+      missingParams = [...missingParams, bodyParam];
+    }
+  });
+
+  if (missingParams.length > 0) {
+    const errorMsg = `The ${missingParams} is required to create a video. Please include ${missingParams} and try again.`;
+    const error = new Error(errorMsg);
+    error.httpStatusCode = 400;
+    return next(error);
+  }
+  if (presentParams.length == props.length) {
+    return next();
+  }
+};
```

Comment:
This is legitimately really cool logic, nice work here man. Also, honestly didn't know you could use the spread operator on arrays.

Reply:
🍠  🤘🏻 sweet! 🤘🏻 🍠